documentation/decisions/adr/00NN-frontend-testing.md

```diff
@@ -0,0 +1,98 @@
+- **Related Issue:** [183](https://github.com/HHS/grants-equity/issues/183) <!-- RECOMMENDED -->
+- **Deciders:** Sammy Steiner, Lucas Brown, Billy Daly, Andy Cochran <!-- REQUIRED -->
+- **Tags:** ADR <!-- OPTIONAL -->
+
+## Context and Problem Statement
+
+Front-end testing is essential in maintaining a stable and healthy codebase, creating UI interfaces without regression, and an important part of the developer workflow.
+
+## Decision Drivers <!-- RECOMMENDED -->
+
+- **Robust:** Chosen testing frameworks should have features that offer diverse ways of verifying front-end codebase functionality, reducing the need for intensive manual testing
+- **Well-maintained:** Accessible tooling is well-maintained by owners and keeps up with current ecosystems in which it will be integrated.
+- **Ease of use:** Achieving high coverage should be attainable. ICs should be able to onboard with the tooling and execute in a reasonable time frame to maintain that coverage.
+- **Fast:** Running tests shouldn't take ages.
+
+## Options Considered
+
+## Unit Testing
+- Jest
+- Mocha
+
+### Pros and Cons of the Options <!-- OPTIONAL -->
+#### [Jest](https://jestjs.io/)
+
+- **Pros**
+  - Lightweight, well-supported and documented testing solution
+  - Compatible with TypeScript, Node, React
+  - Snapshot capabilities to track UI changes
+  - Code coverage reporting that can be integrated into [Github Actions](https://github.com/marketplace/actions/jest-coverage-report)
+  - Already integrated in Next.js front-end template
+  - Built-in mocking & auto-mocking
+- **Cons**
+  - Reports that snapshot capabilities are not for complicated UIs
+  - Auto-mocking reduces performance
+
+#### [Mocha](https://mochajs.org/)
+
+- **Pros**
+  - Has integration and end-to-end testing capabilities
+  - More robust in functionality, offering a wider array of capabilities and helpers
+  - Test coverage reporting available
+  - Well-documented and widely used in Node.js ecosystems
+- **Cons**
+  - Not as lightweight as alternative, requiring integrating several other libraries
+  - Does not support snapshot testing
+  - Slower than Jest
+
+## Visual, Interactive & Component Testing
+- Storybook
+- Cypress
+
+### Pros and Cons of the Options <!-- OPTIONAL -->
+#### [Storybook](https://storybook.js.org)
+
+- **Pro**
+  - Build reusable UI components and pages in isolation, mocking external dependencies and business logic
+  - Visually verify changes and debug on multiple device formats easily in the interface
+  - Stories can be imported into Jest or other testing frameworks
+  - Covers multiple testing types, including visual, interaction, component and accessibility
+  - Compatible with many helpful addons to assist in building front-ends faster
+  - Component testing addon uses Jest & Playwright under the hood for speed
+  - Snapshot capabilities to address DOM regressions
+  - Enhances collaboration between engineers and other team members such as designers
```

Comment:
This point can't be overstated. Nava has found Storybook invaluable for collaboration and documentation of our patterns. It's been used not only by engineers, designers, and product managers, but stakeholders also use Storybook to preview changes and reference existing component/pages/content.